### PR TITLE
Fix incorrect JSON being produced for metadata output

### PR DESCRIPTION
**Pull request by triggerNZ**

### CODE REVIEW

File: core/src/main/scala/commbank/coppersmith/MetadataOutput.scala [hunk 1]
```diff
@@ -60,7 +60,7 @@ object MetadataOutput {
       "valueType" -> jString(genericValueTypeToString(md.valueType)),
       "typesConform" -> jBool(oConforms.isDefined)
     ).nospaces
-  }, lst => s"[${lst.mkString(",")}}]")
+  }, lst => s"[${lst.mkString(",")}]")
```
Comment (laurencer) [inline core/src/main/scala/commbank/coppersmith/MetadataOutput.scala]: Why are you doing manual serialisation here and not simply constructing a `Seq[Json]` and using the pretty-printer for that?

Comment (triggerNZ) [inline core/src/main/scala/commbank/coppersmith/MetadataOutput.scala]: Good question. Essentially to satisfy the `MetadataPrinter` signature, which handles more than json. Let me have a play with generalising this to arbitrary types.

Comment (laurencer) [inline core/src/main/scala/commbank/coppersmith/MetadataOutput.scala]: Can we also add a simple spec (or case class) that describes the output of the JSON format. Right now I think its simply a number of JSON arrays serialised each on a new line. These should be joined together in a single flattened array, an array which contains arrays, or an object with each array as properties.

It'd be nice to know what the semantic differences are or what the structure tells you.

**Comment by laurencer** [comment]
LGTM 👍 - might want to clarify the JSON structure but that can come later if necessary.

**Comment by triggerNZ** [comment]
Cool. I will write some docs explaining the JSON structure in another PR

File: core/src/test/scala/commbank/coppersmith/MetadataOutputSpec.scala [hunk 2]
```diff
@@ -78,4 +79,15 @@ object MetadataOutputSpec extends Specification with ScalaCheck with JsonMatcher
...
+
+  def listProducesValidJson = {
+    import MetadataOutput._
+    val metadataList = List(
+      Metadata[Customer, Integral]("ns", "feature1", "feature1", Discrete),
+      Metadata[Customer, Str]("ns", "feature2", "feature2", Discrete)
+    )
+
+    val generatedJson = JsonObject.combiner(metadataList.map(md => JsonObject.fn(md, None)))
+    argonaut.Parse.parse(generatedJson).isRight === true
```
Comment (dkristian) [inline core/src/test/scala/commbank/coppersmith/MetadataOutputSpec.scala]: I realise it's already merged, but for future reference,
```scala
argonaut.Parse.parse(generatedJson) must beRight
```
would be more idiomatic specs2 code. Matching the value in the `Right` would be even more complete.

**Pull request merged**